### PR TITLE
[RuntimeDyld][Windows] Allocate space for dllimport things.

### PR DESCRIPTION
**Pull request by al45tair**
We weren't taking account of the space we require in the stubs for things that are dllimported, and as a result we could hit the assertion failure for running out of stub space. Fix that.

rdar://133473673

### CODE REVIEW

**Comment by al45tair** [comment]
This is a cherry pick of https://github.com/swiftlang/llvm-project/pull/9071.

**Comment by al45tair** [comment]
Looks to me like the build passed just fine, but the attempt to print the `ccache`/`sccache` stats failed at the end?

**Comment by al45tair** [comment]
@lhames Any chance you could review this upstream PR? Also, I don't have merge rights and I also don't see how to kick the PR testing apparatus into retrying the build here (the failures don't seem to be anything to do with the PR).

File: llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.cpp [hunk 1]
```diff
@@ -119,4 +119,14 @@ bool RuntimeDyldCOFF::isCompatibleFile(const object::ObjectFile &Obj) const {
...
 }
 
+bool RuntimeDyldCOFF::relocationNeedsDLLImportStub(
+    const RelocationRef &R) const {
+  object::symbol_iterator Symbol = R.getSymbol();
+  Expected<StringRef> TargetNameOrErr = Symbol->getName();
+  if (!TargetNameOrErr)
+    return false;
+
+  return TargetNameOrErr->startswith(getImportSymbolPrefix());
```
Comment (bnbarham) [inline llvm/lib/ExecutionEngine/RuntimeDyld/RuntimeDyldCOFF.cpp:129]:
```suggestion
  return TargetNameOrErr->starts_with(getImportSymbolPrefix());
```

**Pull request merged**